util/microservice-sdk/js/lib/kafka/Producer.js

```diff
@@ -0,0 +1,257 @@
+  *
+  * @param {config} config the consumer Producer.
+  * It is an object with the following properties:
+  * - "producer.flush.timeout.ms": Timeout in ms to flush the librdkafka internal queue,
+  *    sending all messages
+  * - "producer.poll.internval.ms": Polls the producer on this interval,
+  *    handling disconnections and reconnection. Set it to 0 to turn it off.
+  * - "producer.connect.timeout.ms":  Timeout  in ms  to connect
+  * - "producer.disconnect.timeout.ms":  Timeout  in ms  to disconnect
+  * - kafka: an object with specific properties for the node-rdkafka consumer. For more details,
+  * see: https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+  */
+  constructor(config) {
+    // configuration
+    this.config = config || {};
+
+    // kafka consumer configuration
+    this.config.kafka = this.config.kafka || {};
+
+    this.config['producer.flush.timeout.ms'] = (
+      this.config['producer.flush.timeout.ms'] || PRODUCER_FLUSH_TIMEOUT_MS
+    );
+
+    this.config['producer.poll.internval.ms'] = (
+      this.config['producer.poll.internval.ms'] || PRODUCER_POLL_INTERNVAL_MS
+    );
+
+    this.config['producer.connect.timeout.ms'] = (
+      this.config['producer.connect.timeout.ms'] || PRODUCER_CONNECT_TIMEOUT_MS
+    );
+
+    this.config['producer.disconnect.timeout.ms'] = (
+      this.config['producer.disconnect.timeout.ms'] || PRODUCER_DISCONNECT_TIMEOUT_MS
+    );
+
+    logger.debug('Creating a new Kafka producer...', TAG);
+
+
+    this.isReady = false;
+    this.producer = new Kafka.Producer(this.config.kafka);
+
+    if (this.isDeliveryReportEnabled()) {
+      // This will set a callback to be executed everytime a message is published.
+      const resolveOnDeliveryReportBind = Producer.resolveOnDeliveryReport.bind(this);
+      this.producer.on('delivery-report', resolveOnDeliveryReportBind);
+    }
+
+    this.producer.setPollInterval(this.config['producer.poll.internval.ms']);
+
+    logger.debug('... a Kafka producer was successfully created.', TAG);
+  }
+
+  /**
+  * Check if the delivery report is active
+  *
+  * To use the event 'delivery-report', you must set request.required.acks to 1 or -1 in topic
+  * configuration and dr_cb (or dr_msg_cb if you want the report to contain the message payload)
+  * to true in kafka options.
+  *
+  * @private
+  */
+  isDeliveryReportEnabled() {
+    return this.config.kafka.dr_cb || this.config.kafka.dr_msg_cb;
+  }
+
+  /**
+   * Connects the Producer to a Kafka cluster.
+   *
+   * This function will wait for 5 seconds in order to successfully connect to
+   * a broker instance. If there is any problem, then it will time out or
+   * node-rdkakfa will emit a event.error event, both of them triggering a
+   * rejection of the returned promise.
+   *
+   * @returns { Promise } A promise which will be resolved if this Producer is
+   * successfully connected to a Kafka broker, or reject if any errors occur.
+   */
+  connect() {
```

Comment:
the lib producer connect function expects some MetaOptions, can we expect them, as a parameter, or in the configuration, and if not provided the default is undefined.
What do you think?

Reply:
On line 137 instead of undefined?

Reply:
In fact, I don't know if there are other parameters
https://github.com/Blizzard/node-rdkafka#methods

Reply:
https://docs.confluent.io/2.0.0/clients/librdkafka/classRdKafka_1_1Producer.html#ab90a30c5e5fb006a3b4004dc4c9a7923
I think at this moment we don't need worry about, but I think It's a good point that you noted.